board.js

```diff
@@ -273,15 +300,80 @@ export default class Board {
             if (!minDistSquaredSum || distSquaredSum < minDistSquaredSum) {
               hasLineOfSight = true;
               minDistSquaredSum = distSquaredSum;
+              minSourcePoint = distSquaredPointToCell(p0, {x: targetX, y: targetY});
               sourceCorner = c0;
               targetCorners = [c1, c2];
             }
+            else if (distSquaredSum === minDistSquaredSum) {
+              let distSourcePoint = distSquaredPointToCell(p0, {x: targetX, y: targetY});
+              if (!minSourcePoint || distSourcePoint < minSourcePoint) {
+                minSourcePoint = distSourcePoint;
+                sourceCorner = c0;
+                targetCorners = [c1, c2];
+              }
+            }
+          }
+        }
+      }
+    }
+    return {
+      hasLineOfSight,
+      sourceCorner,
+      targetCorners,
+    };
+  }
+
+  checkLineOfSightFigures(
+    sourceCells: Array<Point>,
+    targetCells: Array<Point>,
+    optionsIn: ?LineOfSightOptions,
+  ): LineOfSightFiguresResult {
+    let options = optionsIn || defaultLineOfSightOptions();
+    options.cellsToIgnore = _.unionWith(
+      options.cellsToIgnore,
+      sourceCells,
+      targetCells,
+      _.isEqual);
+    let hasLineOfSight = false;
+    let sourceCell = null;
+    let targetCell = null;
+    let sourceCorner = null;
+    let targetCorners = null;
+    let minDistSquaredSum = null;
+    for (let i = 0; i < sourceCells.length; i++) {
+      for (let j = 0; j < targetCells.length; j++) {
+        let c0 = sourceCells[i];
+        let c1 = targetCells[j];
+        let result = this.checkLineOfSight(
+          c0.x,
+          c0.y,
+          c1.x,
+          c1.y,
+          options,
+        );
+        if (result.hasLineOfSight) {
+          invariant(result.sourceCorner, 'no source corner for LoS result');
```

Comment:
you added these for flow right?

there are 2 ways to improve this
1/ use nullthrows inline instead of adding an invariant statement
2/ change the definition of LineOfSightResult to be a bool and maybe object instead of bool and bunch of fields.

I would use 1 for now and we can do 2 later or never